chanfitness/revenue.go

```diff
@@ -0,0 +1,52 @@
+package chanfitness
+
+import (
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/lightningnetwork/lnd/lnwire"
+)
+
+// attributeIncoming is the percentage of fees that are attributed to the
+// incoming channel in a forwarding event. The remaining 1- attributeIncoming
+// percent is attributed to the outgoing channel.
+var attributeIncoming = 0.5
+
+// calculateRevenue calculates the revenue for a channel for the set of events
+// provided. It distributes fees between incoming and outgoing channels
+// involved in htlc forwards according the the value set in attributeIncoming.
+func calculateRevenue(chanID lnwire.ShortChannelID,
+	events []channeldb.ForwardingEvent) lnwire.MilliSatoshi {
+
+	// Revenue is the total revenue earned by this channel in millisatoshis.
+	// This value is accumulated as a float then converted to millisatoshis so
+	// that we can minimize the amount of rounding from float -> int.
+	var revenue float64
```

Comment:
Alternatively, we can used fixed point arithmetic (working in `mSAT`) all the way throughout.